web/documentserver-example/ruby/app/models/file_model.rb

```diff
@@ -254,8 +254,10 @@ def get_history
           obj["created"] = change ? change["created"] : nil
           obj["user"] = change ? change["user"] : nil
 
+          diff_path = [hist_dir, (i - 1).to_s, "diff.zip"].join(File::SEPARATOR)
           prev = histData[(i - 2).to_s]  # get the history data from the previous file version
-          dataObj["previous"] = is_enable_direct_url == true ? {  # write key and url information about previous file version with optional direct url
+          # write key and url information about previous file version with optional direct url
+          dataObj["previous"] = (is_enable_direct_url == true && File.exist?(diff_path)) ? {
```

Comment:
`File.exist?(diff_path)` should be checked in `dataObj["changesUrl"] = `(272 line)

Reply:
> `File.exist?(diff_path)` should be checked in `dataObj["changesUrl"] = `(272 line)

Fixed it